cmd/discovery/cmd.go

```diff
@@ -4,61 +4,123 @@ package main
 import (
 	"encoding/json"
 	"flag"
+	"fmt"
+	"log"
+	"maps"
+	"net/url"
 	"os"
+	"slices"
 
 	"github.com/viam-modules/viamrtsp/viamonvif"
+	"github.com/viam-modules/viamrtsp/viamonvif/device"
 	"go.viam.com/rdk/logging"
 )
 
+// Config for the disovery command.
+type Config struct {
```

Comment:
This command now takes a config file that allows the caller to specify both the credentials (multiples) they want to do discovery with as well as allows a user to specify hosts (ip:port) which they would like to try to detect rtsp streams on using onvif as if discovery had returned those hosts.

This is intended to assist in troubleshooting in the future if we need to test if a given host supports onvif but doesn't support WS-Discovery in the manner our code expects.